tower-balance/src/pool.rs

```diff
@@ -71,14 +73,23 @@ where
                     return Ok(Async::NotReady);
                 }
 
+                tracing::trace!(
+                    pool.services = self.services,
+                    message = "decided to add service to loaded pool"
+                );
                 try_ready!(self.maker.poll_ready());
+                tracing::trace!("making new service");
                 // TODO: it'd be great if we could avoid the clone here and use, say, &Target
                 self.making = Some(self.maker.make_service(self.target.clone()));
             }
         }
 
         if let Some(mut fut) = self.making.take() {
             if let Async::Ready(s) = fut.poll()? {
+                tracing::trace!(
+                    pool.services = self.services,
```

Comment:
The event when deleting a service occurs _after_ the service is deleted and `self.services` is incremented, while this one occurs before. I think we should move this to after the increment so we report the number of services after creating the new service.